flink-end-to-end-tests/test-scripts/test_mesos_streaming_heap_state.sh

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+################################################################################
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+set -o pipefail
+
+source "$(dirname "$0")"/common.sh
+source "$(dirname "$0")"/common_mesos_docker.sh
+
+DURATION=10
+OUTPUT_LOCATION="${TEST_DATA_DIR}/out/tmp_out"
+TEST_PROGRAM_JAR=$END_TO_END_DIR/flink-cli-test/target/PeriodicStreamingJob.jar
```

Comment:
I don't like that we are using a test job that is currently explicitly made for a CLI test. What are our options here:
- Change module name of `PeriodStreamingJob`.
- Use `DataStreamAllroundTestProgram`.
- ???

Reply:
I prefer to add config "SEQUENCE_GENERATOR_SRC_MAX_ELEMENTS" to `DataStreamAllroundTestProgram`, WDYT?

Reply:
I prefer to change the module name of `PeriodStreamingJob`. This job is very versatile, and should not be limit to CLI test. WDYT?